File: YeChanSong_코딩테스트_연습/DFS_BFS/네트워크.py
import sys
sys.setrecursionlimit(10000)

def dfsall(n):
    global visited
    cnt = 0
    for i in range(n):
        if visited[i] == False:
            dfs(i)
            cnt +=1
    return cnt

def dfs(idx):
    global visited, adj
    
    visited[idx] = True
    
    for i in range(len(adj[idx])):
        if i == idx:
            continue
        if adj[idx][i] == 1 and visited[i] == False:
            dfs(i)
    
def solution(n, computers):
    answer = 0
    global visited, adj
    adj = computers
    visited = list(False for i in range(n))
    answer = dfsall(n)
    return answer

File: YeChanSong_코딩테스트_연습/DFS_BFS/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = dfsAll(n,computers);
        return answer;
    }
    public int dfsAll(int n, int[][] adj){
        int cnt=0;
        visited = new boolean[n];
        for(int i=0;i<n;i++){
            if(!visited[i]){
                dfs(i, adj);
                cnt++;
            }
        }
        return cnt;
    }
    public void dfs(int idx, int[][] adj){
        visited[idx] = true;
        for(int i=0;i<adj.length;i++){
            if(adj[idx][i]==1 && !visited[i]){
                dfs(i,adj);
            }
        }
    }
}